### DISCUSSION TITLE
Sampling interval on create monitor item

### DISCUSSION THREAD

**Discussion by Kallelele** [FreeOpcUa/opcua-asyncio#1011] [Q&A]
Hi, I see a need for keeping track of the sampling interval for monitored items.
A monitored item with different sampling interval than `self.parameters.RequestedPublishingInterval` can be created by using: 
`async def create_monitored_items(self, monitored_items) -> List[Union[int, ua.StatusCode]]:`

I think that it would be good to verify this against SamplingInterval in the request with RevisedSamplingInterval in case there is a missmatch and to set samplingInterval to RevisedSamplingInterval with modify monitored items.

What do you think?

**Comment by AndreasHeine** [comment]
Server or Client?

**Reply by Kallelele** [reply]
As Client

**Reply by AndreasHeine** [reply]
so you would like to have an extra call if the server does not accept the requested interval?

**Reply by Kallelele** [reply]
Basically yes, if RevisedSampleInterval in `results = await self.server.create_monitored_items(params)` differ with SampleInterval in the request.
I would like to call Modify monitored item. As it is now you will not get the feedback that SampleInterval was incorrect.

I would like to add this call as well as a parameter for SampleInterval to this class `SubscriptionItemData`

**Reply by AndreasHeine** [reply]
why not read the attribute "minimumSamplingInterval" of the Variable you like to sample before requesting something from which you don't know if it is to small?

i mean a log message would be nice if that happens!
but trying to modify it again? -> do you have a usecase in mind?
the spec says that if the requested interval is to small the server uses the smallest he can!

> The actual sampling interval that the Server will use.This value is based on a number of factors, including capabilities of the underlying system. The Server shall always return a revisedSamplingInterval that is equal or higher than the requested samplingInterval. If the requested samplingInterval is higher than the maximum sampling interval supported by the Server, the maximum sampling interval is returned.

**Reply by Kallelele** [reply]
Ah, that attribute is useful, Thanks.
Ah, then it is unnecessary to send a modify request :)
